### DISCUSSION TITLE
Boolean attributes set to TRUE being deleted from JSON

### DISCUSSION THREAD

**Discussion by pbougie** [tinacms/tinacms#4375] [General]
I have entries in a global JSON document that when saved, the boolean attributes set to `true` are being deleted which breaks my build. My field configuration is as follows:
```
{
  type: "boolean",
  name: "archived",
  label: "Archived ?",
}
``` 
When set to `false`, the attribute is omitted when saving and that is fine.
However, when the attribute is set to `true`, it is now deleting previously set attributes.

Am I configuring the attribute wrong?

**Comment by logan-anderson** [comment]
Hello @pbougie I was not able to reproduce this behaviour. Can you provide a link to your github repo or create one? 

Thanks!

**Comment by pbougie** [comment]
@logan-anderson I've investigated further. I have 38 entries where `"archived": true` is set. Only 6 get deleted when saving the file. This used to work until I upgraded TinaCMS 4 days ago from version `1.5.21` to `1.5.26`.

Entries with the following do not get deleted:
```
{
    ...
    "archived": true,
    "other": ""
}
```

However, entries with the following do get deleted:
```
{
    ...
    "partners": [],
    "archived": true
}
```

Maybe it's because it's the last attribute in the entry? Or it follows an empty array? I've been able to reproduce it in dev mode. Unfortunately, I can't share the repo because it's a private client repo. If the above does not help, let me know and I'll see if I can reproduce it in a new project.

**Reply by logan-anderson** [reply]
I tried making it the last item in the array and that did not work. If you can reproduce in another project I can take a closer look.

**Reply by pbougie** [reply]
@logan-anderson I reduced the project as much as I could to reproduce the problem. You can find it here: https://github.com/pbougie/tinacms-discussion-4375 Check the README for instructions.

**Reply by logan-anderson** [reply]
Thanks I was able to reproduce here is a quick video of what a found and a temporary work around while I work on a fix: https://www.loom.com/share/86184f22fcfd49e79c8d02a3d529f27d

**Reply by logan-anderson** [reply]
Fixed by: https://github.com/tinacms/tinacms/pull/4381

**Comment by logan-anderson** [comment]
Going to close this as it is fixed by https://github.com/tinacms/tinacms/pull/4381

**Comment by pbougie** [comment]
@logan-anderson Thanks for looking into this and explaining with the video. Appreciated.
